src/tests/ftest/util/dfuse_utils.py

```diff
@@ -126,32 +128,64 @@ def remove_mount_point(self):
         """Remove dfuse directory
         Raises:
             CommandFailure: In case of error deleting directory
+
+        Try once with a simple rmdir which should succeed, if this
+        does not then try again with rm -rf, but still raise an error
         """
         # raise exception if mount point not specified
         if self.mount_dir.value is None:
             raise CommandFailure("Mount point not specified, "
                                  "check test yaml file")
 
-        dir_exists, _ = general_utils.check_file_exists(
+        dir_exists, clean_nodes = general_utils.check_file_exists(
             self.hosts, self.mount_dir.value, directory=True)
         if dir_exists:
+
+            target_nodes = self.hosts
```

Comment:
Should this be a copy of `self.hosts`, as it may be modified by `target_nodes.remove(clean_nodes)`?

Reply:
Good catch.